tests/_benderjs/ckeditor/static/tools.js

```diff
@@ -1103,11 +1104,26 @@
 						throw new Error( 'Test named "' + specificTestName + '" already exists' );
 					}
 
-					outputTests[ specificTestName ] = ( function( testName, editorName ) {
-						return function() {
-							inputTests[ testName ]( bender.editors[ editorName ], bender.editorBots [ editorName ] );
-						};
-					} )( testName, editorName );
+					// We are creating new editor instance to isolate each test case (#1696).
```

Comment:
It works fine, however there could be one optimalization applied. The fact that this method is based on `bender.editors` means that when `isolateTests=true` we create new editor instance for each test base on `bender.editors`.

However, bender itself already created editor instances based on `bender.editors` (it happens before test file is executed). So we end up here with few (ofc depends on `bender.editors` content) not used editor instance. I would suggest alternating this method in a way that for the first iteration it uses already created editor instances and then it creates new ones.